### PR TITLE
Add recipe for codespell

### PR DESCRIPTION
**Pull request by dopplershift**
Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by dopplershift** [comment]
License not currently packaged because it is not included in the source distribution. PR has been made upstream (codespell-project/codespell#1230)

**Comment by dopplershift** [comment]
Looks like this passed all builds and is ready to go. cc: @conda-forge/help-python

File: recipes/codespell/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - python
+
+test:
+  commands:
+    - codespell --version
+
+about:
+  home: https://github.com/codespell-project/codespell
+  license: GPL-2.0
+  license_family: GPL
```
Comment (chrisburr) [inline recipes/codespell/meta.yaml]: GPL-2.0 requires the licence text to be included with all copies of the software:
```suggestion
  license_family: GPL
  license_file: LICENSE
```
If it's not in the tarball you can download/include it in the recipe directory.

**Comment by dopplershift** [comment]
Closing in favor of #9187 (because I accidentally deleted my fork and now can't update this PR.

**Pull request closed without merging**